### PR TITLE
Add method to open an arbitrary subdirectory in native explorer

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

File: osu.Framework/Platform/Storage.cs [hunk 1]
```diff
@@ -123,6 +123,11 @@ public virtual Storage GetStorageForDirectory(string path)
         /// <summary>
         /// Opens a native file browser window to the root path of this storage.
         /// </summary>
-        public abstract void OpenInNativeExplorer();
+        public virtual void OpenInNativeExplorer() => OpenPathInNativeExplorer(string.Empty);
```
Comment (frenzibyte) [inline osu.Framework/Platform/Storage.cs:126]: Doesn't need to be virtual IMO, `OpenPathInNativeExplorer` should be enough to handle that.
```suggestion
        public void OpenInNativeExplorer() => OpenPathInNativeExplorer(string.Empty);
```

Comment (peppy) [reply osu.Framework/Platform/Storage.cs:126]: See the osu!-side PR.

**Comment by smoogipoo** [comment]
Why do we need two virtual methods? Couldn't you expose only the `PathIn` version and override that?

**Comment by peppy** [comment]
Ah, my bad. Should be fine without.

**Pull request merged**